matscipy/dislocation.py

```diff
@@ -2162,64 +2175,91 @@ def __init__(self, unit_cell, alat, C11, C12, C44, axes, burgers,
 
         Parameters
         ----------
-        unit_cell : unit cell to build the dislocation configuration
-        alat : lattice constant
+        a : lattice constant OR cubic bulk atoms object to build the dislocation configuration
         C11 : elastic constants
         C12
         C44
-        axes : cell axes (b is normally along z direction)
-        burgers : burgers vector of the dislocation
-        unit_cell_core_position : dislocation core position in the unit cell
-                                  used to shift atomic positions to
-                                  make the dislocation core the center
-                                  of the cell
-        parity
+        symbol : Chemical symbol used to construct
+
+
+        Useful Dislocation Properties
+        ------------------------------
+        alat : float
+            lattice parameter (in A)
+        unit_cell : ase.atoms.Atoms object
+            bulk cell used to generate dislocations
+        axes : np.array of float
+            unit cell axes (b is normally along z direction)
+        burgers_dimensionless : np.array of float
+            burgers vector of the dislocation class, in a dimensionless alat=1 system
+        burgers : np.array of float
+            burgers vector of the atomistic dislocation
+        unit_cell_core_position : np.array of float
+            dislocation core position in the unit cell used to shift atomic positions to
+            make the dislocation core the center of the cell
+        parity : np.array
         glide_distance : distance to the next equivalent
                          core position in the glide direction
         n_planes : int
             number of non equivalent planes in z direction
         self_consistent : float
             default value for the displacement calculation
+        crystalstructure : str
+            Name of basic structure defining atomic geometry ("fcc", "bcc", "diamond")
         """
 
-        self.unit_cell = unit_cell.copy()
-        self.alat = alat
+        # Create copies of immutable class attributes
+        # Prevents side effects when E.G. changing burgers vector of an instance
+        # Also makes changing cls.var and instance.var work as expected for these variables
+        self.axes = self.axes.copy()
+        self.burgers_dimensionless = self.burgers_dimensionless.copy()
+        self.unit_cell_core_position_dimensionless = self.unit_cell_core_position_dimensionless.copy()
+        self.parity = self.parity.copy()
+
         self.C11 = C11
         self.C12 = C12
         self.C44 = C44
+        
+        self.alat, self.unit_cell = _validate_disloc_inputs(a, symbol, self.axes, self.crystalstructure, self.pbc)
```

Comment:
rename to `_validate_cubic_cell()` and move out of dislocation.py? Perhaps create a new top-level `matscipy.utils` module and import from there to here.